### PR TITLE
Fix custom repositories import so it supports "list of values" custom column types  [fixes SCI-2024]

### PR DESCRIPTION
**Pull request by ZmagoD**
https://biosistemika.atlassian.net/browse/SCI-2024

### CODE REVIEW

File: app/utilities/repository_import_parser/importer.rb [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    end
+
+    def new_repository_row(row)
+      RepositoryRow.new(name: row[@name_index],
+                        repository: @repository,
+                        created_by: @user,
+                        last_modified_by: @user)
+    end
+
+    def import_to_database(row_cell_values)
```
Comment (okriuchykhin) [inline app/utilities/repository_import_parser/importer.rb]: How can we extend this later?

File: app/utilities/repository_import_parser/importer.rb [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    def run
+      get_columns
+      return check_for_duplicate_columns if check_for_duplicate_columns
+      import_rows!
+    end
+
+    private
+
+    def get_columns
```
[the remark this answers is not in the document]
Comment (okriuchykhin) [reply app/utilities/repository_import_parser/importer.rb]: maybe rename it to fetch_columns or load_columns ?

File: app/utilities/repository_import_parser/repository_cell_value_resolver.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    def initialize(column, user, repository)
+      @column = column
+      @user = user
+      @repository = repository
+    end
+
+    def get_value(value, record_row)
+      return unless @column
+      send("new_#{@column.data_type.underscore}", value, record_row)
```
[the remark this answers is not in the document]
Comment (okriuchykhin) [reply app/utilities/repository_import_parser/repository_cell_value_resolver.rb]: can you please fix this?

File: app/utilities/repository_import_parser/repository_cell_value_resolver.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+# this class is used to resolve the column data_type and assign
+# it to the repository_row
+module RepositoryImportParser
+  class RepositoryCellValueResolver
+
```
[the remark this answers is not in the document]
Comment (okriuchykhin) [reply app/utilities/repository_import_parser/repository_cell_value_resolver.rb]: can you please fix this?

[the remark this answers is not in the document]
Comment (okriuchykhin) [reply app/utilities/repository_import_parser/repository_cell_value_resolver.rb]: can you please fix this?

File: app/utilities/repository_import_parser/importer.rb (hunk 1 above)
Comment (ZmagoD) [reply app/utilities/repository_import_parser/importer.rb]: you make a switch statement or I don't know do we need this to be extendable from addons?

File: app/utilities/repository_import_parser/repository_cell_value_resolver.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (ZmagoD) [reply app/utilities/repository_import_parser/repository_cell_value_resolver.rb]: is not working with public send

File: app/utilities/repository_import_parser/importer.rb (hunk 1 above)
Comment (okriuchykhin) [reply app/utilities/repository_import_parser/importer.rb]: Yes, to add a data type from addon

**Pull request merged**